src/visitors/semantic_analysis_visitor.cpp

```diff
@@ -66,5 +68,14 @@ void SemanticAnalysisVisitor::visit_destructor_block(const ast::DestructorBlock&
     /// -->
 }
 
+void SemanticAnalysisVisitor::visit_unit(const ast::Unit& node) {
+    /// <-- This code is for check 3
+    if (node.get_name()->get_value().empty()) {
+        logger->warn("SemanticAnalysisVisitor:: An unit cannot be created without name.");
```

Comment:
Since the mod file translation fails with this I'm wondering whether we should print this message as `error` instead of `warning`.

Reply:
done